### ISSUE TITLE
UTF-8 characters wider than 1 byte mess up tables

### ISSUE THREAD

**Issue by zoidby** [woefe/ytcc#62]
For 1 byte long characters, the tables look correct:

```
ASCII Title          | 
abcde                |
```
The `|` is at the same position.

But more complex UTF-8 characters mess up the space calculation:
```
ASCII Title          |
Title with 😁         |
```
It seems, due to the emoji needing 2 bytes, but actually only displaying as one byte, the calculation for the spaces is wrong. The  same applies to German umlauts and other scripts like Japanese.

**Comment by woefe** [comment]
The length calculation should see emojis as a single character, unless your Python installation is messed up somehow. Are you using UTF-8 locales? What´s the output of the `locale` command?

The issue you are experiencing is probably a combination of your terminal emulator and font setup. German umlauts work flawlessly for me, but some emojis are printed two characters wide.

The `|` in the output of following command is probably not properly aligned, correct?
```
echo -e "😀 |\na |"
```

I think that the length calculation should handle wide Unicode characters as two cells, which is different from Python's len() on strings. Not sure what the best practices are.

**Comment by zoidby** [comment]
You are right, the display of characters like emojis are double width, not their code. And umlauts work fine.

My locales are
```
> locale
LANG=en_US.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC=de_DE.UTF-8
LC_TIME=de_DE.UTF-8
LC_COLLATE=C
LC_MONETARY=de_DE.UTF-8
LC_MESSAGES=C
LC_PAPER=de_DE.UTF-8
LC_NAME=de_DE.UTF-8
LC_ADDRESS=de_DE.UTF-8
LC_TELEPHONE=de_DE.UTF-8
LC_MEASUREMENT=de_DE.UTF-8
LC_IDENTIFICATION=de_DE.UTF-8
LC_ALL=
```

But i guess the echo is expected to be wrong since there is no calculation for needed spaces happening. So i don’t think this has to do with my terminal setup and needs to be fixed in ytcc.

**Comment by woefe** [comment]
So, I found an example that seems to calculate the width correctly.
```
echo -e "😀 |a\n1 |a" | column -s '|' -t -o "|"
```
I try do dig through https://github.com/karelzak/util-linux/blob/master/text-utils/column.c to see how they calculate the width. Nevertheless, it still comes down to whether your terminal emulator displays emojis as one or two cells. https://bugs.launchpad.net/ubuntu/+source/gnome-terminal/+bug/1665140

**Comment by woefe** [comment]
The column command uses [wcwidth()](https://pubs.opengroup.org/onlinepubs/007904975/functions/wcwidth.html). I'll use [jquast/wcwidth](https://github.com/jquast/wcwidth/blob/b29897e5a1b403a0e36f7fc991614981cbc42475/wcwidth/wcwidth.py#L140) to implement it in ytcc

**Comment by woefe** [comment]
Fixed in 54d67c4

**Comment by zoidby** [comment]
Works. Very awesome, thank you!
